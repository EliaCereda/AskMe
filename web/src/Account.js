import React, {Component} from "react";

class Account extends Component {
    render() {
        const isLoggedIn = (this.props.currentUser !== null);

        if (isLoggedIn) {
            return <p>
                Welcome back, <strong>{this.props.currentUser}</strong>! {}
                <a href="#" onClick={() => this.props.onLogin(null)}>Log Out</a>
            </p>
        } else {
            return <LoginForm onLogin={this.props.onLogin}/>;
        }
    }
}

export default Account;

class LoginForm extends Component {
    constructor(props) {
        super(props);

        this.state = {username: null};

        this.handleChange = this.handleChange.bind(this);
        this.handleSubmit = this.handleSubmit.bind(this);
    }

    handleChange(event) {
        this.setState({username: event.target.value});
    }

    handleSubmit(event) {
        this.props.onLogin(this.state.username);

        event.preventDefault();
    }

    render() {
        return (
            <div className="LoginForm">
                <form onSubmit={this.handleSubmit}>
                    <input type="text" placeholder="Username" onChange={this.handleChange} required/>
                    <input type="submit" value="Log In"/>
                </form>
            </div>
        )
    }
}
