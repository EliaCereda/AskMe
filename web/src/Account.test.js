import React from 'react';
import Account from "./Account";

it('renders correctly', () => {
  let tree;

  tree = shallow(<Account currentUser="Giorgia" />);
  expect(tree).toMatchSnapshot();

  tree = shallow(<Account currentUser={null} />);
  expect(tree).toMatchSnapshot();
});
