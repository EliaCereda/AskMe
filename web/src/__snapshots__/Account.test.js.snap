// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders correctly 1`] = `
<p>
  Welcome back, 
  <strong>
    Giorgia
  </strong>
  ! 
  <a
    href="#"
    onClick={[Function]}
  >
    Log Out
  </a>
</p>
`;

exports[`renders correctly 2`] = `<LoginForm />`;
